### PR TITLE
FOGL-7432 override logging error in python logger class

### PR DESCRIPTION
**Pull request by ashish-jabble**
Note - We need to revisit logger error/exception messages to get traceback in multiline via another JIRA. First, we will track down Python core and then Admin API.

### CODE REVIEW

**Comment by praveen-garg** [comment]
Remove those loop based statements in exception handler blocks? To test and clean up the base branch code

**Review by MarkRiddoch — DISMISSED** [review]
Looks good to me

File: python/fledge/common/logger.py [hunk 1]
```diff
@@ -8,7 +8,11 @@
 import os
 import subprocess
 import logging
+import inspect
```
Comment (praveen-garg) [inline python/fledge/common/logger.py:11]: remove?

File: python/fledge/common/logger.py [hunk 2]
```diff
@@ -181,10 +202,37 @@ def get_logger(self, logger_name: str):
...
         self.add_handlers(_logger, [syslog_handler, console_handler])
         _logger.propagate = False
+
+        @wraps(_logger.error)
+        def error(msg, *args, **kwargs):
+            """Case: When we pass exception in error and having different args
+            For example:
+            a) _logger.error(ex)
+            b) _logger.error(ex, "Failed to add data.")
+            """
```
Comment (praveen-garg) [inline python/fledge/common/logger.py:220]: """ Override error logger to print multi-line traceback and error string with newline"

case/example part can be moved to if block, as given in the else part.

File: python/fledge/common/logger.py [hunk 3]
```diff
@@ -181,10 +202,37 @@ def get_logger(self, logger_name: str):
...
+            if isinstance(msg, Exception):
+                trace_msg = traceback.format_exception(msg.__class__, msg, msg.__traceback__)
+                if args:
+                    trace_msg[:0] = ["{}\n".format(args[0])]
+                [__logging_error(line.strip('\n')) for line in trace_msg]
+            else:
+                """Case: When we pass string in error
+                For example:  
+                a) _logger.error(str(ex))
+                b) _logger.error("Failed to add data for key: {} along with error:{}".format("Test", str(ex)))
```
Comment (praveen-garg) [inline python/fledge/common/logger.py:230]: a) _logger.error("Failed to log audit trail entry")
           b) _logger.error("Failed to log audit trail entry '{}' \n{}".format(code, str(ex))) # \n will result the str(ex) to be printed in next line

**Review by praveen-garg — DISMISSED** [review]
Tested okay with injecting a simulated exception.

Let's  do a manual review of syslogs with some make based system test run & also backup/restore/north python special logs

**Pull request merged**